core/src/main/java/com/datastrato/gravitino/storage/kv/KvEntityStore.java

```diff
@@ -280,20 +280,18 @@ public boolean delete(NameIdentifier ident, EntityType entityType, boolean casca
                 ident, subEntities);
           }
 
-          for (byte[] prefix : subEntityPrefix) {
-            transactionalKvBackend.deleteRange(
-                new KvRange.KvRangeBuilder()
-                    .start(prefix)
-                    .startInclusive(true)
-                    .end(Bytes.increment(Bytes.wrap(prefix)).get())
-                    .build());
-          }
+          // Remove id-name mapping;
+          unbindNameAndId(ident, entityType);
 
-          deleteAuthorizationEntitiesIfNecessary(ident, entityType);
           return transactionalKvBackend.delete(dataKey);
         });
   }
 
+  private void unbindNameAndId(NameIdentifier ident, EntityType entityType) throws IOException {
+    String identNameToIdKey = generateKeyForMapping(ident, entityType, nameMappingService);
+    nameMappingService.unbindNameAndId(identNameToIdKey);
+  }
+
```

Comment:
Should we still delete these entries?
`unbindNameAndId `is the same of deleteRange(prefix_min, prefix_max)

Reply:
> Should we still delete these entries?

No, we can't use a background thread to do the work just as JVM GC

> unbindNameAndId is the same of deleteRange(prefix_min, prefix_max)

`unbindNameAndId` will make the entity and it's sub-entities inaccessible to users and they are equal logically.

Reply:
Will it be possible that other same-name entity still use this name?

Reply:
Not possible as each name is attched with context, for example, the key '1/2/name' represents a specific entity under a metalake (1) and catalog (2).

Reply:
Do you need more tests to cover?